iothub/device/src/ClientOptions.cs

```diff
@@ -4,12 +4,12 @@
 namespace Microsoft.Azure.Devices.Client
 {
     /// <summary>
-    /// Options that allow configuration of the device client instance during initialization.
+    /// Options that allow configuration of the device or module client instance during initialization.
     /// </summary>
     public class ClientOptions
```

Comment:
We need to rename this to avoid the name conflcit with, and it needs to inherit from the Azure.Core.ClientOptions.